src/utransport.rs

```diff
@@ -250,16 +273,23 @@ pub trait UTransport: Send + Sync {
     /// ```
     async fn register_listener(
```

Comment:
I'm thinking the reason you only have the default `UNIMPLEMENTED` error on `receive` and `register_listener` but not the `send`. Is this because developers can choose to implement either `receive` or `register_listener`?

Reply:
exactly, or both, actually